extensions/azure/fcc-node-directory-cosmos/src/main/java/org/eclipse/dataspaceconnector/catalog/node/store/cosmos/CosmosFederatedCacheNodeStoreExtension.java

```diff
@@ -0,0 +1,133 @@
+/*
+ *  Copyright (c) 2020, 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.catalog.node.store.cosmos;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import org.eclipse.dataspaceconnector.catalog.node.store.cosmos.model.FederatedCacheNodeDocument;
+import org.eclipse.dataspaceconnector.catalog.spi.FederatedCacheNodeDirectory;
+import org.eclipse.dataspaceconnector.common.string.StringUtils;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.security.Vault;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+import java.util.ArrayList;
+import java.util.Set;
+
+/**
+ * Provides a persistent implementation of the {@link FederatedCacheNodeDirectory} using CosmosDB.
+ */
+public class CosmosFederatedCacheNodeStoreExtension implements ServiceExtension {
+    /**
+     * The setting for the CosmosDB account name
+     */
+    @EdcSetting
+    private static final String COSMOS_ACCOUNTNAME_SETTING = "edc.cosmos.account.name";
+    /**
+     * The setting for the name of the database where FederaredCatalogNode will be stored
+     */
+    @EdcSetting
+    private static final String COSMOS_DBNAME_SETTING = "edc.cosmos.database.name";
+    @EdcSetting
+    private static final String COSMOS_PARTITION_KEY_SETTING = "edc.cosmos.partitionkey";
+
+    private static final String DEFAULT_PARTITION_KEY = "dataspaceconnector";
+    private static final String CONTAINER_NAME = "federatedcatalognode";
+    private static final String NAME = "CosmosDB Federated Catalog Node Store";
+
+    private Monitor monitor;
+
+    @Override
+    public Set<String> provides() {
+        return Set.of(FederatedCacheNodeDirectory.FEATURE);
+    }
+
+    @Override
+    public Set<String> requires() {
+        return Set.of();
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        monitor = context.getMonitor();
+        monitor.info(String.format("Initializing %s extension...", NAME));
+
+        // configure cosmos db
+        var cosmosAccountName = context.getSetting(CosmosFederatedCacheNodeStoreExtension.COSMOS_ACCOUNTNAME_SETTING, null);
```

Comment:
what do you think about creating a POJO that contains all configuration items? It would get the `ServiceExtensionContext` passed in to the CTor